modules/systemd.nix

```diff
@@ -73,22 +75,87 @@ let
 
   servicesStartTimeoutMs = builtins.toString cfg.servicesStartTimeoutMs;
 
-  unitType =
-    unitKind:
-    with types;
-    let
-      primitive = oneOf [
-        bool
-        int
-        str
-        path
+  unitBaseType =
+    unitKind: mod:
+    types.submodule {
+      freeformType =
+        with types;
+        let
+          primitive = oneOf [
+            bool
+            int
+            str
+            path
+          ];
+        in
+        attrsOf (attrsOf (either primitive (listOf primitive)))
+        // {
+          description = "systemd ${unitKind} unit configuration";
+        };
+
+      imports = [
+        {
+          options.Unit = {
+            Description = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              example = "My daily database backup";
+              description = "A short human-readable label of the unit.";
+            };
+
+            Documentation = mkOption {
+              type = with types; either str (listOf str);
+              apply = p: if builtins.isList p then p else [ p ];
```

Comment:
This is `lib.toList` (a no-op on lists), but otherwise using `coercedTo str lib.toList (listOf str)` removes the need for `apply`.